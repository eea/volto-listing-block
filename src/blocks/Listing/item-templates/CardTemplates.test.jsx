import React from 'react';
import renderer from 'react-test-renderer';
import {
  DefaultCardLayout,
  LeftImageCardLayout,
  RightImageCardLayout,
  ImageCardLayout,
} from './CardTemplates';

jest.mock('@plone/volto/components', () => ({
  ConditionalLink: () => <div>ConditionalLink</div>,
  UniversalLink: () => <div>UniversalLink:</div>,
}));

describe('DefaultCardLayout', () => {
  it('renders correctly', () => {
    const item = {
      title: 'DefaultCardLayout title',
      description: 'DefaultCardLayout description',
      meta: 'DefaultCardLayout meta',
    };

    const itemModel = {
      hasImage: true,
      hasDate: true,
      hasDescription: true,
      imageOnRightSide: false,
    };

    const component = renderer.create(
      <DefaultCardLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
        imagePosition="left"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('LeftImageCardLayout', () => {
  it('renders correctly', () => {
    const item = {
      title: 'LeftImageCardLayout title',
      description: 'LeftImageCardLayout description',
      meta: 'LeftImageCardLayout meta',
    };

    const itemModel = {
      hasImage: true,
      hasDate: true,
      hasDescription: true,
      imageOnRightSide: false,
    };

    const component = renderer.create(
      <LeftImageCardLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
        imagePosition="left"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('RightImageCardLayout', () => {
  it('renders correctly', () => {
    const item = {
      title: 'RightImageCardLayout title',
      description: 'RightImageCardLayout description',
      meta: 'RightImageCardLayout meta',
    };

    const itemModel = {
      hasImage: true,
      hasDate: true,
      hasDescription: true,
      imageOnRightSide: false,
    };

    const component = renderer.create(
      <RightImageCardLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
        imagePosition="left"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('ImageCardLayout', () => {
  it('renders correctly', () => {
    const item = {
      title: 'ImageCardLayout title',
      description: 'ImageCardLayout description',
      meta: 'ImageCardLayout meta',
    };

    const itemModel = {
      hasImage: true,
      hasDate: true,
      hasDescription: true,
      imageOnRightSide: false,
    };

    const component = renderer.create(
      <ImageCardLayout
        item={item}
        itemModel={itemModel}
        className="my-class"
        imagePosition="left"
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
