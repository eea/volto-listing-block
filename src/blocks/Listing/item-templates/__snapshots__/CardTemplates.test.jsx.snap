// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultCardLayout renders correctly 1`] = `
<div
  className="ui fluid card u-card my-class"
  onClick={[Function]}
>
  <div>
    ConditionalLink
  </div>
  <div
    className="content"
  >
    <div
      className="meta"
    >
      <span
        className="text-right date"
      >
        04 Oct 2023
      </span>
    </div>
    <div
      className="header"
    >
      <div>
        ConditionalLink
      </div>
    </div>
  </div>
</div>
`;

exports[`ImageCardLayout renders correctly 1`] = `
<div
  className="ui fluid card u-card my-class"
  onClick={[Function]}
>
  <div>
    ConditionalLink
  </div>
</div>
`;

exports[`LeftImageCardLayout renders correctly 1`] = `
<div
  className="ui fluid card u-card my-class item-card left-image-card"
  onClick={[Function]}
>
  <div>
    ConditionalLink
  </div>
  <div
    className="content"
  >
    <div
      className="meta"
    >
      <span
        className="text-right date"
      >
        04 Oct 2023
      </span>
    </div>
    <div
      className="header"
    >
      <div>
        ConditionalLink
      </div>
    </div>
  </div>
</div>
`;

exports[`RightImageCardLayout renders correctly 1`] = `
<div
  className="ui fluid card u-card my-class item-card right-image-card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="meta"
    >
      <span
        className="text-right date"
      >
        04 Oct 2023
      </span>
    </div>
    <div
      className="header"
    >
      <div>
        ConditionalLink
      </div>
    </div>
  </div>
  <div>
    ConditionalLink
  </div>
</div>
`;
